src/rotation.rs

```diff
@@ -48,6 +48,57 @@ impl<T> Angle<T>
     }
 }
 
+impl<T> Angle<T>
+where T: Rem<Output=T>
+    + Sub<Output=T>
+    + Add<Output=T>
+    + Zero
+    + FloatConst
+    + PartialOrd
+    + Copy
+{
+    /// Returns this angle in the [0..2*PI[ range.
+    pub fn positive(&self) -> Self {
+        let two_pi = T::PI() + T::PI();
+        let mut a = self.radians % two_pi;
+        if a < T::zero() {
+            a = a + two_pi;
+        }
+        Angle::radians(a)
+    }
+
+    /// Returns this angle in the ]-PI..PI] range.
+    pub fn signed(&self) -> Self {
+        Angle::pi() - (Angle::pi() - *self).positive()
+    }
+}
+
+impl<T> Angle<T>
+where T: Float {
+    /// Returns (sin(self), cos(self)).
+    pub fn sin_cos(self) -> (T, T) {
+        self.radians.sin_cos()
+    }
+}
+
+impl<T> Angle<T>
+where T: Zero {
+    pub fn zero() -> Self { Angle::radians(T::zero()) }
+}
+
+impl<T> Angle<T>
+where T: FloatConst + Add<Output=T> {
+    pub fn pi() -> Self { Angle::radians(T::PI()) }
+
+    pub fn two_pi() -> Self { Angle::radians(T::PI() + T::PI()) }
+
+    pub fn frac_pi_2() -> Self { Angle::radians(T::FRAC_PI_2()) }
```

Comment:
Does the `Angle` have a semantic of radians? I thought of it as an abstracted opaque angle, in which case having `pi` in the name here is leaking that abstraction.

Reply:
`Angle` exposes its internals as radians, I didn't try to abstract out the unit as much as just make the type more convenient (and explicit), while keeping some helpers to deal with degrees without too much hassle.

I think that it's best if radians to are the first class citizen and I would rather be transparent about it than build an abstraction. But it doesn't go against picking names that are more "units agnostic" when convenient. Not sure what those could be for these constants, though. Do you have ideas?